[Proposal] Symfony Guide

Jibbarth opened the pull request:
After [trying Pest on a Symfony application](https://xn--jibbarth-d1a.fr/gist/c45838ede5cde76b2856530d32df7754/), here is a slight guide to use Pest on Symfony App. 

I thought the main key is to focus on how to use WebTestCase as it's the most common used tests, and I based my examples to reproduce the test created by `php bin/console make:functional-test`

The second part is how to create helpers for all tests, like retrieve a client instance or the container. 

WDYT ?

Jibbarth committed:
Add a symfony guide

Jibbarth committed:
Add link to symfony guide

Jibbarth committed:
Clean sentence about ide autocomplete

nunomaduro commented on source/docs/guides/symfony.md:
```diff
@@ -0,0 +1,147 @@
+
+Now, in this guide, we are going to transform a Symfony test into a Pest test.
+
+A basic functionnal test in Symfony look like the following:
```
```suggestion
A basic functional test in Symfony looks like the following:
```

nunomaduro commented on source/docs/guides/symfony.md:
```diff
@@ -0,0 +1,147 @@
+
+You can now remove the `uses(..)` in all your tests in `Controller` directory.
+
+## Helpers
```
I think this all section should be removed.

nunomaduro replied:
For now.

Jibbarth replied:
Even `function container` and `function client()` ?

nunomaduro commented on source/docs/guides/symfony.md:
````diff
@@ -0,0 +1,147 @@
+}
+```
+
+With Pest, the same test could be write like this:
````
```suggestion
With Pest, the same test could be written like this:
```

nunomaduro commented on source/docs/guides/symfony.md:
```diff
@@ -0,0 +1,147 @@
+uses(\Symfony\Bundle\FrameworkBundle\Test\WebTestCase::class);
+
+it('test something', function() {
+    static::createClient()->request('GET', '/');
```
Can't you chain this? Like:
```php
static::createClient()->request('GET', '/')
    ->assertResponseIsSuccessful()
    ->assertSelectorTextContains('h1', 'Hello World');
```

Jibbarth replied:
No, request return a [Crawler](https://github.com/symfony/dom-crawler/blob/master/Crawler.php) and assertions are void methods.

nunomaduro commented on source/docs/guides/symfony.md:
````diff
@@ -0,0 +1,147 @@
+```php
+<?php
+
+uses(\Symfony\Bundle\FrameworkBundle\Test\WebTestCase::class);
````
Can you start by creating a `Pest.php` with this content, and then talk about the test? I think it would make it cleaner.

Jibbarth replied:
I can, but my idea was to introduce one test as single tests, then introduce the Pest power :muscle: 

Let me rethink it :wink:

Jibbarth committed:
Typos

Co-authored-by: Nuno Maduro <enunomaduro@gmail.com>

Jibbarth committed:
Merge branch 'master' into feature/symfony-guide

Jibbarth committed:
Remove helpers part

Jibbarth committed:
Create Pest.php first

nunomaduro commented:
@AlexMartinFR Can you take a look?

alexmartinfr committed:
Update symfony.md

- Changed the Pest test example to use the test() function, to better reflect the original Symfony example
- Typo corrected

alexmartinfr reviewed (commented):
I changed the Pest test example (now uses test() function), otherwise it does look perfect to me. I say, ready for merge 👍

nunomaduro committed:
feat(symfony-guide): updates docs

nunomaduro commented:
Thank you @Jibbarth
